nextcloud/server
# [WIP] Dont allow reusing usernames

We have cases in which app developers forget to delete user data when a user is being deleted. This could be problematic as future users with the same username could then access this data.

This mitigates this risk a bit by not allowing reusing of usernames. That said, we still need to fix occurrences of these issues by deleting missed data.

# Test plan

- Create user in the UI
- Delete user
- Check table for inserted hashed user ID
- Try to create user with same UID. See it fail.
- Other users can still be created.

There is currently no feedback in the UI but that is due to another bug, which also affects other errors such as the ones by an enforced password policy: https://github.com/nextcloud/server/issues/27115

# TODOs 

- [x] Adjust DI for Manager
- [ ] Fix up code
- [x] Listen to user deletion event and mark user name as used
- [ ] Fixup commits
- [ ] Fix tests that reuse the same UID
